agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -93,15 +95,63 @@ def create_terminal_model(self) -> crocoddyl.ActionModelAbstract:
         """Create the terminal model."""
         pass
 
-    def set_reference_weighted_trajectory(
-        self, reference_weighted_trajectory: list[WeightedTrajectoryPoint]
-    ):
+    def set_reference_weighted_trajectory(self):
         """Set the reference trajectory for the OCP."""
-        if self._ocp_params.use_debug_data:
-            reference_trajectory_points = [
-                el.point for el in reference_weighted_trajectory
+        pass
+
+    def get_nb_iterations(self):
+        """Get max number of solver iterations and qp iterations."""
+        return self._ocp_params.solver_iters, self._ocp_params.qp_iters
+
+    def set_nb_iterations(self, nb_iter, qp_iters):
+        """Set max number of solver iterations and qp iterations."""
+        self._solver.max_qp_iters = qp_iters
+        self._ocp_params.qp_iters = qp_iters
+        self._ocp_params.solver_iters = nb_iter
+
+    def fill_debug_data(self, res, ocp_results):
+        """Fill attributes of dataclass OCPDebugData."""
+        self._debug_data.problem_solved = res
+        self._debug_data.result = ocp_results
+        self._debug_data.kkt_norm = self._solver.KKT
+        self._debug_data.nb_iter = int(self._solver.iter)
+        self._debug_data.nb_qp_iter = int(self._solver.qp_iters)
+        # TODO the keys used for residuals and references should be parameters
+        if len(self._debug_data.residuals) == 0:
+            costs_dict = self._problem.runningModels[
+                0
+            ].differential.costs.costs.todict()
+            names = costs_dict.keys()
+            residuals_names = [
+                name
+                for name in names
+                if isinstance(costs_dict[name].cost.residual, ResidualDistanceCollision)
+            ]  # get only collision distance residuals
+            references_names = [
+                name
+                for name in names
+                if not isinstance(
+                    costs_dict[name].cost.residual, ResidualDistanceCollision
+                )
             ]
-            self._debug_data.references = reference_trajectory_points
+        else:
+            residuals_names = self._debug_data.residuals.keys()
+            references_names = self._debug_data.references.keys()
+
+        self._debug_data.residuals = {name: [] for name in residuals_names}
+        self._debug_data.references = {name: [] for name in references_names}
+        for name in references_names:
+            model = self._problem.runningModels[0]
+            reference = model.differential.costs.costs[name].cost.residual.reference
+            if isinstance(reference, pin.SE3):
+                reference = pin.SE3ToXYZQUAT(reference)
+
+            self._debug_data.references[name].append(reference)
```

Comment:
Why do we limit to the reference of the first element? Is it because we assume the references are shifted and the other reference can be reconstructed?

If that is the case, please add a comment.

Reply:
There should be a comment here then.
From the user side, this comment will not be easy to find. I'm not sure how topics are documented...